### DISCUSSION TITLE
Laravel socialite package Google sign in

### DISCUSSION THREAD

**Discussion by nikospapcom** [laravel/framework#44605] [Q&A]
Hello,

I'm using Laravel Socialite package to sign in with Google and I want to find a way to pass some extra parameters to login (e.g. redirect url) in order to redirect in the correct page after user login

I'm trying something like this by no results:
`Socialite::driver('google')->with(['rd'=>'https://google.com'])->redirect();`

Seems like all parameters in with() method skipped by

Any suggestion about it?

**Comment by kverstae** [comment]
`with()` is used to set custom parameters in the get request to the authentication provider.
What you do here, will just send an extra field `rd` with the request to google. Since they don't know what to do with this, this will just be ignored.
If the only thing you want to do is change the redirect url after login, you can do this in the callback route which gets called when the authentication provider gives you the user. An example of this can be found here: https://laravel.com/docs/9.x/socialite#authentication-and-storage

**Reply by nikospapcom** [reply]
Yes I already did this one but the original problem is that I want to implement dynamic redirect back, e.g user logged in from www.example.com/test-page?parameter=test I want to return back to the same page

Do you have something in your mind how I can do this?

**Reply by kverstae** [reply]
One thing I can think of is using the session to store this information. 
So in the redirect route, you set a session variable to the url you want to redirect to and then on the callback route you read that session variable again.
